app/src/main/java/org/simple/clinic/registration/phone/RegistrationPhoneScreenController.kt

```diff
@@ -181,4 +181,13 @@ class RegistrationPhoneScreenController @Inject constructor(
                 .startWith(Observable.just({ ui: Ui -> ui.hideAnyError() }, { ui: Ui -> ui.showProgressIndicator() }))
         )
   }
+
+  private fun showLoggedOutOnThisDeviceDialog(events: Observable<UiEvent>): Observable<UiChange> {
+    return events
+        .ofType<RegistrationPhoneScreenCreated>()
+        .flatMapSingle { userSession.loggedInUser().firstOrError() }
+        .filterAndUnwrapJust()
+        .filter { loggedInUser -> loggedInUser.loggedInStatus == UNAUTHORIZED }
+        .flatMap { Observable.just { ui: Ui -> ui.showLoggedOutOnThisDeviceDialog() } }
```

Comment:
This could be simplified to just this: 
`map { { ui: Ui -> ui.showLoggedOutOnThisDeviceDialog() } }`

Reply:
Also, can we extract if user is `UNAUTHORIZED` to `UserSession` itself ?
The controller need not know what unauthorised means.
Wdyt?

Reply:
> This could be simplified to just this:
> `map { { ui: Ui -> ui.showLoggedOutOnThisDeviceDialog() } }`

Oh yeah, this is doable. I don't know why I missed this. 😅

Reply:
> Also, can we extract if user is `UNAUTHORIZED` to `UserSession` itself ?
> The controller need not know what unauthorised means.
> Wdyt?

This is a good idea, we have duplicated this code in quite a few places. I'll do this.

Reply:
Done.